### PR TITLE
Add RDS DBInst W/Del Protection Disab query #1054

### LINKED ISSUE #1054: Add RDS DBInstance With Deletion Protection Disabled query for AWS CloudFormation
### Platform
CloudFormation

### Provider
AWS 

### Description
RDS DBInstance should have deletion protection set to true

### PR DESCRIPTION
**Pull request by joelcarvalhocheckmarx**
RDS DBInstance should have deletion protection set to true

Closes #1054

### CODE REVIEW

File: assets/queries/cloudFormation/rds_db_instance_with_deletion_protection_disabled/metadata.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "rds_db_instance_with_deletion_protection_disabled",
+  "queryName": "RDS DBInstance With Deletion Protection Disabled",
+  "severity": "LOW",
+  "category": null,
+  "descriptionText": "RDS DBInstance should have deletion protection set to true",
+  "descriptionUrl": "https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-rds-database-instance.html#cfn-rds-dbinstance-deletionprotection"
+}
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_db_instance_with_deletion_protection_disabled/metadata.json:8]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/rds_db_instance_with_deletion_protection_disabled/metadata.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "rds_db_instance_with_deletion_protection_disabled",
+  "queryName": "RDS DBInstance With Deletion Protection Disabled",
+  "severity": "LOW",
+  "category": null,
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_db_instance_with_deletion_protection_disabled/metadata.json:5]: `Backup and Disaster Recovery`

File: assets/queries/cloudFormation/rds_db_instance_with_deletion_protection_disabled/query.rego [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  not exists_dp
+  
+      result := {
+                "documentId": 		input.document[i].id,
+                "searchKey":        sprintf("Resources.%s.Properties.DeletionProtection", [name]),
+                "issueType":		   "MissingAttribute",
+                "keyExpectedValue": sprintf("Resources.%s.Properties.DeletionProtection is defined", [name]),
+                "keyActualValue": 	sprintf("Resources.%s.Properties.DeletionProtection is undefined", [name])
+              }
+}
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_db_instance_with_deletion_protection_disabled/query.rego:34]: [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/rds_db_instance_with_deletion_protection_disabled/test/negative.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    Properties:
+      DBInstanceClass: !Ref DBInstanceType
+      SourceDBInstanceIdentifier: !Ref SourceDBInstanceIdentifier
+      SourceRegion: !Ref SourceRegion
+      DeletionProtection: true
+      KmsKeyId: !Ref MyKey
+Outputs:
+  InstanceId:
+    Description: InstanceId of the newly created RDS Instance
+    Value: !Ref MyDBSmall
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_db_instance_with_deletion_protection_disabled/test/negative.yaml:39]: Remove trailing whitespace.

[As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: assets/queries/cloudFormation/rds_db_instance_with_deletion_protection_disabled/test/positive.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    Type: "AWS::RDS::DBInstance"
+    Properties:
+      DBInstanceClass: !Ref DBInstanceType
+      SourceDBInstanceIdentifier: !Ref SourceDBInstanceIdentifier
+      SourceRegion: !Ref SourceRegion
+      KmsKeyId: !Ref MyKey
+Outputs:
+  InstanceId:
+    Description: InstanceId of the newly created RDS Instance
+    Value: !Ref MyDBSmall
```
Comment (rogeriopeixotocx) [inline assets/queries/cloudFormation/rds_db_instance_with_deletion_protection_disabled/test/positive.yaml:80]: * Change the Reference string to `MyDBSmall1`

* Remove trailing whitespace.

* [As a good POSIX practice](https://unix.stackexchange.com/a/18789) add end-of-file newline

File: go.sum [hunk 6]
```diff
@@ -31,6 +31,7 @@ github.com/Microsoft/hcsshim v0.8.7/go.mod h1:OHd7sQqRFrYd3RmSgbgji+ctCwkbq2wbEY
 github.com/OneOfOne/xxhash v1.2.2/go.mod h1:HSdplMjZKSmBqAxg5vPj2TmRDmfkzw+cTzAElWljhcU=
```
Comment (rogeriopeixotocx) [inline go.sum:31]: Why the changes in `go.sum`?

Try not to commit those changes when developing queries unless really necessary.

**Review by rogeriopeixotocx — CHANGES_REQUESTED** [review]
Almost good

**Comment by rogeriopeixotocx** [comment]
LGTM ✅

**Review by ruigomescx — APPROVED** [review]
